opendevin/controller/command_manager.py

```diff
@@ -2,43 +2,51 @@
 
 from opendevin.observation import CmdOutputObservation
 from opendevin.sandbox.sandbox import DockerInteractive
+from opendevin.sandbox.e2b_sandbox import E2BSandbox
+
 
 class CommandManager:
-    def __init__(self, dir: str, container_image: str | None = None,):
+    def __init__(
+        self,
+        dir: str,
+        container_image: str | None = None,
+    ):
         self.directory = dir
-        self.shell = DockerInteractive(id="default", workspace_dir=dir, container_image=container_image)
+        # self.shell = DockerInteractive(
+        #     id="default", workspace_dir=dir, container_image=container_image
+        # )
+        self.shell = E2BSandbox()
 
     def run_command(self, command: str, background=False) -> CmdOutputObservation:
-        if background:
-            return self._run_background(command)
-        else:
-            return self._run_immediately(command)
+        print(f"Running command in background: {background}")
+        return self._run_immediately(command)
+        # if background:
```

Comment:
we'll definitely need background support before merging this--otherwise the agents will get stuck.

Does e2b support long-running commands? E.g. can the agent run `node server.js` and then `curl localhost:3000`?

Reply:
Yes, that's possible. Each sandbox has it's own URL and we automatically allow you to connect to a port when it opens. We support HTTP request and Websockets at the moment. This way you can connect for example to a running Nextjs app that you start with `next dev` and see how OpenDevin makes changes in real-time.